### ISSUE TITLE
Bad Insurgency performance (compared to native)

### ISSUE THREAD

**Issue by mradermaxlol** [flathub/com.valvesoftware.Steam#83]
I don't know if it's the only case of such performance degradation, but when I try to play Insurgency from flatpak'ed Steam I get ~50% performance of a native installation (from Arch repos, that is). It could be driver-related as well, as I use AMDGPU on my GCN1.0 GPU (Radeon R7 370), but I guess the sandboxing itself has something to do with it.

My system is Arch w/ testing repositories, linux-zen 4.16.9, mesa 18.0.4 (multilib as well) & flatpak 0.11.7. The issue has been present since the first days of the package being available at Flathub.

**Comment by nanonyme** [comment]
Iirc the runtime includes another build or Mesa which may be significantly older and potentially slower. Try running bash inside Flatpack and checking with usual means whether you even have hardware-accelerated GL

**Comment by mradermaxlol** [comment]
@nanonyme 
> Try running bash inside Flatpack

Sorry for a (probably) dumb question, but how does one do that?

**Comment by TingPing** [comment]
It should have 18.0.2 now which probably isn't worse off than 18.0.4 on the host.

**Comment by mradermaxlol** [comment]
@nanonyme so how do I run `bash` inside flatpak?

**Comment by TingPing** [comment]
This is probably more useful:
```
flatpak install flathub org.freedesktop.GlxInfo
flatpak run org.freedesktop.GlxInfo
```

**Comment by mradermaxlol** [comment]
@TingPing thanks!
```
$ flatpak run org.freedesktop.GlxInfo | grep direct
direct rendering: Yes
```

```
$ flatpak run org.freedesktop.GlxInfo | grep OpenGL
OpenGL vendor string: X.Org
OpenGL renderer string: AMD Radeon (TM) R7 300 Series (PITCAIRN / DRM 3.23.0 / 4.16.9-1-zen, LLVM 6.0.0)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 18.0.2
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.0 Mesa 18.0.2
OpenGL shading language version string: 1.30
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 18.0.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
OpenGL ES profile extensions:
```

So I guess it's not the runtime issue. Once again, maybe it's because of the whole sandboxing thing?

**Comment by TingPing** [comment]
Being in the sandbox alone should have nearly no effect on performance.

**Comment by mradermaxlol** [comment]
Okay... Any other guesses?

**Comment by nexfwall** [comment]
Maybe, Steam is using it's internal runtime from Ubuntu 12.04 for games?

P.S. Just passing by.

**Comment by mradermaxlol** [comment]
@nexfwall well, when running it native I don't use my system libs but Steam's builtin instead (those Ubuntu 12.04, probably, yeah), so Steam stuff is probably identical.

**Comment by nexfwall** [comment]
> @nexfwall <https://github.com/nexfwall> well, when running it native I
> don't use my system libs but Steam's builtin instead (those Ubuntu 12.04,
> probably, yeah), so Steam stuff is probably identical.
>
Steam automatically disabling runtime if ALL required dependencies are
found in the system. And AUR package can easily pull them by deps.

I can suggest checking the output of the Steam launcher by launching it in
terminal. Maybe even by this command: `bash -x "$(which steam)"`, to see
all that Steam launcher does before starting app.

>

**Comment by nanonyme** [comment]
Didn't @TingPing disable that logic in the Flatpak way back?

**Comment by nexfwall** [comment]
I don't know about that.

But you should also start checking Flatpacked games vs native. Because it
could be something bigger than just Steam.

**Comment by TingPing** [comment]
@nanonyme It uses default Steam behaviour. It would use a few libs from the flatpak runtime like 64bit  SDL2 probably.

**Comment by nanonyme** [comment]
I'm referring to STEAM_RUNTIME_PREFER_HOST_LIBRARIES=0 which you added in the shell script towards beginning of this repo

**Comment by TingPing** [comment]
That is no longer used.

**Comment by nanonyme** [comment]
Right,  assuming Flatpak 1.0.0 or newer?

**Comment by mradermaxlol** [comment]
@nexfwall I can assure you I _don't_ have any of those native runtime deps installed. Will try comparing native and flatpak'ed output nevertheless.

**Comment by mradermaxlol** [comment]
@nexfwall I've compared the output with `steamcommandname 2&> {native log, flatpak log}`. Raw files were 1.2Gb each 0_0 but that was probably because of **../common/html/chrome_ipc_server.cpp (501) : Assertion Failed: !"Invalid PID for wait"** line that was repeated a few thousand times probably, so I `grep`'d the files with the following command:
`grep -vwF "../common/html/chrome_ipc_server.cpp (501) : Assertion Failed: !\"Invalid PID for wait\"" native.log > natnew.log`
for both files, which you can download below.

Hope this helps! By the way, I've noticed that in (flatpak here) Insurgency's menu nicknames are blank (probably due to missing fonts or smth) just like the key tips in-game. But that's probably another issue ;)

[natnew.log](https://github.com/flathub/com.valvesoftware.Steam/files/2028479/natnew.log)
[fnew.log](https://github.com/flathub/com.valvesoftware.Steam/files/2028480/fnew.log)

**Comment by mradermaxlol** [comment]
It probably has something to do with the initial tests of the game with flatpak'd Steam, according to https://github.com/flathub/com.valvesoftware.Steam/wiki/Tested-Games - see "Insurgency" there. I'll rename the issue accordingly.

By the way, the output of `flatpak --gl-drivers`:
```
$ flatpak --gl-drivers
default
host
```

**Comment by nanonyme** [comment]
Is there any difference now? Mesa was upgraded as part of Freedesktop runtime 18.08 upgrade

**Comment by nanonyme** [comment]
We currently disable shader cache by default due to regressions, you could also see if turning it on helpshttps://github.com/flathub/com.valvesoftware.Steam/wiki/Frequently-asked-questions

**Comment by pablodav** [comment]
> I'm referring to STEAM_RUNTIME_PREFER_HOST_LIBRARIES=0 which you added in the shell script towards beginning of this repo

could be good to have this and also the org.freedesktop.GlxInfo in wiki?

**Comment by pablodav** [comment]
I have tested it, just don't work now. adding notes to #135

**Comment by nanonyme** [comment]
Should work now. Closing as obsolete. Please re-open if persists.
